syntax = "proto3";

package bookstore;

service BookService {
  rpc ListBooks (Empty) returns (ListBooksResponse);
  rpc CreateBook (CreateBookRequest) returns (CreateBookResponse);
  rpc UpdateBook (UpdateBookRequest) returns (Empty);
  rpc DeleteBook (DeleteBookRequest) returns (Empty);
}

message Empty {}

message Book {
  int32 id = 1;
  string title = 2;
  string author = 3;
  double price = 4;
}

message ListBooksResponse {
  repeated Book books = 1;
}

message CreateBookRequest {
  string title = 1;
  string author = 2;
  double price = 3;
}

message CreateBookResponse {
  int32 id = 1;
}

message UpdateBookRequest {
  int32 id = 1;
  string title = 2;
  string author = 3;
  double price = 4;
}

message DeleteBookRequest {
  int32 id = 1;
}
